### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by neon60**

### CODE REVIEW

**Comment by neon60** [comment]
I have to test the rocm-smi build and install, because my changes effecting that too.

**Comment by neon60** [comment]
@saadrahim please check this PR and give some feedbacks. I think we have to split up the readme file.

**Comment by dmitrii-galantsev** [comment]
@saadrahim @samjwu Please have a look. The docs changes are a little over my head :)

**Review by MathiasMagnus — APPROVED** [review]
The resulting docs look fine.

I think the difference in doc style may warrant a spinoff issue to unify the style of the Python API and the C++ API (moving the C++ API over to Breathe as opposed to the current doxysphinx).

File: docs/python_api.rst [hunk 1]
```diff
@@ -0,0 +1,265 @@
+====================
+Python API Reference
+====================
+
+This chapter describes the rocm_smi Python module API.
```
Comment (samjwu) [inline docs/python_api.rst]:
```suggestion
This chapter describes the ROCm SMI Python module API.
```

File: docs/c++_tutorials.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
+====================
+C++ Tutorials
+====================
+
+This chapter is the rocm_smi C++ api tutorials.
```
Comment (samjwu) [inline docs/c++_tutorials.rst]:
```suggestion
This chapter contains the ROCm SMI C++ API tutorials.
```

**Review by samjwu — APPROVED** [review]
Looks good to me.

Had a couple suggestions: I see the README spells the project name as ROCm SMI

**Comment by neon60** [comment]
@dmitrii-galantsev I had to split up the python_smi_tools/rsmiBindings.py file to two separate files:
- python_smi_tools/rsmiBindings.py
- python_smi_tools/rsmiBindingsInit.py

Is it okay with you and your team? Can you check the generated documentation? Do you have any feedbacks?
We would like to merge this in, one day :)

File: docs/sphinx/requirements.txt [hunk 3]
```diff
@@ -116,7 +116,7 @@ requests==2.28.2
     # via
     #   pygithub
     #   sphinx
-rocm-docs-core[api_reference]>=0.30.3
+rocm-docs-core[api_reference]>=0.31.0
```
Comment (charis-poag-amd) [inline docs/sphinx/requirements.txt:119]: ~~Should this also be 30.3? Or is there a reason for bumping up to 31.0?~~
Disregard - large enough change to bump up for. 👍

Comment (neon60) [reply docs/sphinx/requirements.txt:119]: @charis-poag-amd Can you merge this PR in here? I would like to avoid uploading via Gerrit?

**Review by dmitrii-galantsev — APPROVED** [review]
Thank you for your work! Sorry it took a while to respond. I will approve here and merge internally. Should propagate in the next day or 2!

**Comment by dmitrii-galantsev** [comment]
Merged: https://github.com/ROCm/rocm_smi_lib/commit/50a079af0f059ddfafe27e908f4878d759891eca

**Pull request closed without merging**